cmd/transportRequestUploadSOLMAN.go

```diff
@@ -53,31 +53,80 @@ func transportRequestUploadSOLMAN(config transportRequestUploadSOLMANOptions, te
 
 	// Error situations should be bubbled up until they reach the line below which will then stop execution
 	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
-	err := runTransportRequestUploadSOLMAN(&config, &solman.UploadAction{}, telemetryData, utils)
+	err := runTransportRequestUploadSOLMAN(&config, &solman.UploadAction{}, telemetryData, utils, commonPipelineEnvironment)
 	if err != nil {
 		log.Entry().WithError(err).Fatal("step execution failed")
 	}
 }
 
-func runTransportRequestUploadSOLMAN(config *transportRequestUploadSOLMANOptions, action solman.Action, telemetryData *telemetry.CustomData, utils transportRequestUploadSOLMANUtils) error {
+func runTransportRequestUploadSOLMAN(config *transportRequestUploadSOLMANOptions,
+	action solman.Action,
+	telemetryData *telemetry.CustomData,
+	utils transportRequestUploadSOLMANUtils,
+	commonPipelineEnvironment *transportRequestUploadSOLMANCommonPipelineEnvironment) error {
+
 	action.WithConnection(solman.Connection{
 		Endpoint: config.Endpoint,
 		User:     config.Username,
 		Password: config.Password,
 	})
-	action.WithChangeDocumentID(config.ChangeDocumentID)
-	action.WithTransportRequestID(config.TransportRequestID)
+
+	cdID, err := getChangeDocumentID(config, &transportRequestUtils{})
+	if err != nil {
+		return err
+	}
+	action.WithChangeDocumentID(cdID)
+
+	trID, err := getTransportRequestID(config, &transportRequestUtils{})
+	if err != nil {
+		return err
+	}
+	action.WithTransportRequestID(trID)
 	action.WithApplicationID(config.ApplicationID)
 	action.WithFile(config.FilePath)
 	action.WithCMOpts(config.CmClientOpts)
-	err := action.Perform(utils, utils)
+
+	commonPipelineEnvironment.custom.changeDocumentID = cdID
+	commonPipelineEnvironment.custom.transportRequestID = trID
+
+	err = action.Perform(utils, utils)
 
 	if err == nil {
 		log.Entry().Infof("Upload of artifact '%s' to SAP Solution Manager succeeded (ChangeDocumentId: '%s', TransportRequestId: '%s').",
 			config.FilePath,
-			config.ChangeDocumentID,
-			config.TransportRequestID,
+			cdID,
+			trID,
 		)
 	}
 	return err
 }
+
+type itransportRequestUtils interface {
+	FindIDInRange(label, from, to string) (string, error)
+}
+type transportRequestUtils struct {
```

Comment:
What purpose do you get from an empty struct type?

Reply:
the containing method is defined by pointer receiver below.